### ISSUE TITLE
Error when regenerating npm-shrinkwrap 

### ISSUE THREAD

**Issue by jesusbotella** [CartoDB/cartodb#14362]
I noticed that `update-internal-deps` command is failing when regenerating `npm-shrinkwrap.json`.

It throws this error:
```
npm ERR! Problems were encountered
npm ERR! Please correct and try again.
npm ERR! peer invalid: acorn@^6.0.0, required by acorn-dynamic-import@4.0.0
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
```

It can be fixed by installing `acorn@^6.0.0` with `npm i acorn@6.0.0 --save`, but it adds acorn dependency as a first level dependency, which is not what we want (afaik).

**Comment by ramiroaznar** [comment]
@jesusbotella sorry about my lack of understanding, how important is this issue? is it for backend or for front?

**Comment by jesusbotella** [comment]
Sorry, I didn't labeled this issue 🤦‍♂️ 

It's meant for Frontend, it is relatively important when it comes to development because we cannot update CARTO.js version in Builder until we fix this, for example.

But it's better to ask @rjimenezda and assign the proper priority under RT :)

**Comment by rjimenezda** [comment]
## What's going on

The package acorn-node, which is somewhere along the dependencies of the project (browserify > insert-module-globals > acorn-node) recently upgraded some things. They moved to acorn 6 and updated acorn-dynamic-import to 4.0 on [their 1.6.1 version](https://github.com/browserify/acorn-node/commit/26bf15e09c4557f0a25772ec02b44680472cdf92).

acorn-dynamic-import does not have acorn as a dependency, but as a devDependency and peerDependency.

With our current dependencies on Builder, this fails to install on npm3. I don't really understand why yet. 

By manually installing acorn as @jesusbotella mentions, this is solved. But we don't really want that. It's good to have a workaround though.

## Why didn't this happen before

Because prior to v4 of acorn-dynamic-import, acorn was a dependency for them, so it got installed every time. Now it doesn't. That doesn't make a lot of sense, though, as you'll see on the next point.

## Further findings

I've tried making a dummy package with the suspects for this (webpack and browserify are the only ones that require acorn things) but it's able to perform the shrinkwrapping.

This package will always have the following script, just like we do:

```json
"scripts": {
    "update-internal-deps": "rm -rf node_modules && npm install --production --no-optional --no-shrinkwrap && npm dedupe && npm prune && npm shrinkwrap && npm install"
  },
```

This was my first attempt, update-internal-deps works fine
```json
"dependencies": {
    "browserify": "13.0.0",
    "webpack": "4.12.0"
}
```

However, if you add @carto/carto.js, it breaks.
```json
"dependencies": {
    "@carto/carto.js": "4.1.7",
    "browserify": "13.0.0",
    "webpack": "4.12.0"
}
```

Funnily enough it does not break when you point to the repository. 
```json
"dependencies": {
    "internal-carto.js": "CartoDB/carto.js#master",
    "browserify": "13.0.0",
    "webpack": "4.12.0"
}
```

This is very interesting, because the public carto.js package is generated on build, but the dependencies **are the same ones**. 

All is pointing to @carto/carto.js, so naturally I tried removing it from Builder and shrinkwrap works fine. 

I noticed something funny with carto.js (Cartodb/carto.js#2205) and am on the process of fixing it, I don't know if it will affect this, but we'll see

**Comment by rjimenezda** [comment]
Removing browserify-shim from carto.js public version's dependecies (CartoDB/carto.js#2205 seems to fix this problem. Will try to find out why and see if we can do it

**Comment by rjimenezda** [comment]
Turns out browserify-shim is a dependecy of Builder as well, so I am officialy lost. 

This might even be an npm bug, as it doesn't really happen with npm 6. Though handling of peer dependencies might've changed too.

**Comment by ramiroaznar** [comment]
Looks good to me!

**Comment by rjimenezda** [comment]
Closed via #14369
